#!/usr/bin/env python

PACKAGE = 'pandora_costmap'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0,    "An In.....", 50, 0,  100)
#                    name        type   level  description  def min max
gen.add("min_elevation", double_t, 0, "Min height of pc kept", 0.40, 0.0, 5.0);
gen.add("max_elevation", double_t, 0, "Max height of pc kept", 1.20, 0.0, 5.0);

gen.add("inlier_radius", double_t, 0, "Radius of point clusters", 0.1, 0.01, 0.5);
gen.add("min_neighbours_num", int_t, 0, "Minimum num of points in a cluster", 3, 1, 20);

gen.add("leaf_size", double_t, 0, "Voxel's dimension", 0.03, 0.01, 0.1);

exit(gen.generate(PACKAGE, "Filter", "Filter"))
